support/compressxdr/compress_xdr.go

```diff
@@ -3,36 +3,50 @@ package compressxdr
 import (
 	"io"
 
-	"github.com/stellar/go/support/errors"
+	xdr3 "github.com/stellar/go-xdr/xdr3"
 )
 
-const (
-	GZIP = "gzip"
-)
+func NewXDREncoder(compressor Compressor, xdrPayload interface{}) XDREncoder {
+	return XDREncoder{Compressor: compressor, XdrPayload: xdrPayload}
+}
+
+func NewXDRDecoder(compressor Compressor, xdrPayload interface{}) XDRDecoder {
+	return XDRDecoder{Compressor: compressor, XdrPayload: xdrPayload}
 
-type XDREncoder interface {
-	WriteTo(w io.Writer) (int64, error)
 }
 
-type XDRDecoder interface {
-	ReadFrom(r io.Reader) (int64, error)
-	Unzip(r io.Reader) ([]byte, error)
+// XDREncoder combines compression with XDR encoding
+type XDREncoder struct {
+	Compressor Compressor
+	XdrPayload interface{}
 }
 
-func NewXDREncoder(compressionType string, xdrPayload interface{}) (XDREncoder, error) {
-	switch compressionType {
-	case GZIP:
-		return &XDRGzipEncoder{XdrPayload: xdrPayload}, nil
-	default:
-		return nil, errors.Errorf("invalid compression type %s, not supported", compressionType)
+// WriteTo writes the XDR compressed encoded data
+func (e *XDREncoder) WriteTo(w io.Writer) (int64, error) {
```

Comment:
```suggestion
func (e XDREncoder) WriteTo(w io.Writer) (int64, error) {
```
`xdr3.Marshal()` should not modify `e.XdrPayload` so this function does not require a pointer receiver